lib/liquid/block_body.rb

```diff
@@ -45,7 +45,7 @@ def freeze
           end
           tag_name = Regexp.last_match(1)
           markup   = Regexp.last_match(2)
-          unless (tag = registered_tags[tag_name])
+          unless (tag = registered_tag(tag_name, parse_context))
```

Comment:
We should be able to use `parse_context.tags` unconditionally, which would simplify this to

```suggestion
          unless (tag = parse_context.tags[tag_name])
```

we could then get rid of the `registered_tags` method, which used to just be there for monkey patching, but we have since got ridden dependence on it.